### PR TITLE
start of work to add flux executor

### PR DESCRIPTION
**Pull request by Paul-Ferrell**
Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

File: lib/pavilion/schedulers/plugins/flux.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+
+        self._prepare_job_formatter()
+        output = job.sched_log.as_posix()
+        error = job.sched_log.as_posix() + ".err"
+
+        # Generate the flux job
+        # Assume the filename includes a hashbang
+        # flux does not support mem_mb, disk_mb
+        fluxjob = JobspecV1.from_command(
+            command=["/bin/bash", job.kickoff_path.as_posix()],
```
Comment (Paul-Ferrell) [inline lib/pavilion/schedulers/plugins/flux.py:173]: `tasks_per_node` probably already defaults to 1.

File: lib/pavilion/schedulers/plugins/flux.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+            num_tasks=sched_config['tasks_per_node'] or 1,
+        )
+        
+        # A duration of zero (the default) means unlimited
+        fluxjob.duration = sched_config['time_limit'] or 0
+        fluxjob.stdout = output
+        fluxjob.stderr = error
+
+        # This doesn't seem to be used?
```
Comment (Paul-Ferrell) [inline lib/pavilion/schedulers/plugins/flux.py:182]: The kickoff script will end up changing directory anyway, so this won't matter.

File: lib/pavilion/schedulers/plugins/flux.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+        fluxjob.cwd = str(job.tests_path)
+        fluxjob.environment = dict(os.environ)
```
Comment (Paul-Ferrell) [inline lib/pavilion/schedulers/plugins/flux.py:184]: The auto generated kickoff script takes care of environment setup for us, so nothing to see here.

File: lib/pavilion/schedulers/plugins/flux.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+        flux_future = self._fexecutor.submit(fluxjob)
+        
+        job._jobid = str(flux_future.jobid())
+        job._submit_time = time.time()
```
Comment (Paul-Ferrell) [inline lib/pavilion/schedulers/plugins/flux.py:188]: Ok, cool, you've realized you need to convert the job info into a serializable format to save to file. The _flux_future probably won't ever get used (once a job is scheduled in the run process, that's all Pavilion does with the job object other than hit `save()`.

File: lib/pavilion/schedulers/plugins/flux.py [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+            raise ValueError(stderr)
+
+        results = []
+        for section in stdout.split('\n\n'):
+            try:
+                results.append(self._scontrol_parse(section))
+            except (KeyError, ValueError):
+                pass
+
+        return results
```
Comment (Paul-Ferrell) [inline lib/pavilion/schedulers/plugins/flux.py:232]: Not necessary for Flux.

File: lib/pavilion/schedulers/plugins/flux.py (hunk 3 above)
Comment (vsoch) [reply lib/pavilion/schedulers/plugins/flux.py:154]: Maybe, but I think it's important that flux receives all the information that is needed (albeit the same) for consistency.

File: lib/pavilion/schedulers/plugins/flux.py (hunk 5 above)
Comment (vsoch) [reply lib/pavilion/schedulers/plugins/flux.py]: haha yes I know, this is left over from before.

**Comment by vsoch** [comment]
@Paul-Ferrell could you please close this PR so I can re-open as the owner?

**Comment by Paul-Ferrell** [comment]
done

**Comment by vsoch** [comment]
thank you!

**Pull request closed without merging**